lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -1233,11 +1222,36 @@ private synchronized void onConnected(ProtocolMessage message) {
             return;
         }
 
+        //RTN19a
+        sendPendingQueueMessages();
+
         /* indicated connected currentState */
         setSuspendTime();
         requestState(new StateIndication(ConnectionState.connected, error));
     }
 
+    /**
+     * Send all pending messages which are in the queue.
+     * Remove them from the queue once they are sent successfully
+     * Spec: RTN19a
+     */
+    private void sendPendingQueueMessages() {
+        //RTN19a
+        if (pendingMessages.queue != null && !pendingMessages.queue.isEmpty()) {
```

Comment:
Could `isEmpty` check be skipped without any impact? 🤔

Reply:
Indeed both checks are not necessary. So I removed them both